packages/blocks-ui/src/canvas.js

```diff
@@ -18,12 +22,57 @@ const Wrap = props => (
   />
 )
 
+function copyToClipboard(toCopy) {
+  const el = document.createElement(`textarea`);
+  el.value = toCopy;
+  el.setAttribute(`readonly`, ``);
+  el.style.position = `absolute`;
+  el.style.left = `-9999px`;
+  document.body.appendChild(el);
+  el.select();
+  document.execCommand(`copy`);
+  document.body.removeChild(el);
+}
+
+const Copy = ({ toCopy }) => {
+  const [hasCopied, setHasCopied] = useState(false);
+
+  function copyToClipboardOnClick() {
+    if (hasCopied) return;
+
+    copyToClipboard(toCopy);
+    setHasCopied(true);
+
+    setTimeout(() => {
+      setHasCopied(false);
+    }, 2000);
+  }
+
+  return (
+    <IconButton onClick={copyToClipboardOnClick} sx={{ float: "right" }}>
+      {hasCopied ? (
+        <>
+          <Check sx={{ color: "green" }} aria-label="Copied" />
+        </>
+      ) : (
+        <>
+          <Clipboard aria-label="Copy" />
+        </>
+      )}
+    </IconButton>
+  );
+};
+
 export default ({ code, transformedCode, scope, theme }) => {
   const { mode } = useEditor()
 
   if (mode === 'code') {
     return (
       <Wrap>
+        <Copy toCopy={prettier.format(code, {
```

Comment:
Let's initialize the formatted code on line 68 so we don't have to call prettier twice.